### PR TITLE
Renamed platform folder to npu.

### PR DESCRIPTION
**Pull request by anton7811**
The idea is to align real folder content with the SAI-Challenger code.
Platform folder contains sai_npu implementations so it's natural to rename it to "npu".

Signed-off-by: Anton Putria <anton7811@gmail.com>

### CODE REVIEW

**Comment by anton7811** [comment]
Also, I found the SC_PLATFORM variable. But I haven't found any usage of it.

File: docs/porting_guide.md [hunk 1]
```diff
@@ -28,11 +28,11 @@ platform/
...
 ```
 
-The platform folder MAY contain multiple vendors. Each vendor folder MAY contain 1..n ASICs and optional vendor-specific `sai_npu` module. Each ASIC folder MAY contain 1..n targets (either HW device or simulator or emulator) and optional ASIC-specific `sai_npu` module. Each target's folder MUST contain either `Dockerfile` for standalone mode or `Dockerfile.server` for client-server mode or both. Also, the target's folder MAY contain optional `sku/` folder with JSON files that define front panel ports configuration as well as others target-specific files and folders.
+The NPY folder ***may*** contain multiple vendors. Each vendor folder ***may*** contain 1..n ASICs and optional vendor-specific `sai_npu` module. Each ASIC folder ***may*** contain 1..n targets (either HW device or simulator or emulator) and optional ASIC-specific `sai_npu` module. Each target's folder ***must*** contain either `Dockerfile` for standalone mode or `Dockerfile.server` for client-server mode or both. Also, the target's folder ***may*** contain optional `sku/` folder with JSON files that define front panel ports configuration as well as others target-specific files and folders.
```
Comment (elchupanebrej) [inline docs/porting_guide.md:31]: NPY -> NPU

Comment (andriy-kokhan) [inline docs/porting_guide.md:31]: The idea was to follow RFC style - "MUST", "MAY", etc. Please revert.

Comment (anton7811) [reply docs/porting_guide.md:31]: Fixed

Comment (anton7811) [reply docs/porting_guide.md:31]: Fixed

**Review by andriy-kokhan — APPROVED** [review]

**Pull request merged**